From 7f5098d74c5cde99faac27f9962e71a8b09dc5a4 Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Tue, 24 Apr 2018 20:05:22 +0800
Subject: [PATCH] lineagehw: Disable color balance support by default

 * Currently color balance is used for adjusting color temperature if
   it's supported, but QDCM color balance and display mode override each
   other. Obviously we should take good care of display modes.
 * Disable color balance so color temperature will pick color
   calibration as backend, which doesn't conflict with display modes.

Change-Id: I09675682a7a94d7613acf5b366b7e371edd669d3
---
 src/org/lineageos/hardware/ColorBalance.java | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/src/org/lineageos/hardware/ColorBalance.java b/src/org/lineageos/hardware/ColorBalance.java
index 3260bfa..714983e 100644
--- a/src/org/lineageos/hardware/ColorBalance.java
+++ b/src/org/lineageos/hardware/ColorBalance.java
@@ -17,8 +17,6 @@
 
 package org.lineageos.hardware;
 
-import vendor.lineage.livedisplay.V1_0.Feature;
-
 /**
  * Color balance support
  *
@@ -29,16 +27,13 @@ import vendor.lineage.livedisplay.V1_0.Feature;
  */
 public class ColorBalance {
 
-    private static final boolean sHasNativeSupport =
-            LiveDisplayVendorImpl.getInstance().hasNativeFeature(Feature.COLOR_BALANCE);
-
     /**
      * Whether device supports color balance control
      *
      * @return boolean Supported devices must return always true
      */
     public static boolean isSupported() {
-        return sHasNativeSupport;
+        return false;
     }
 
     /**
@@ -48,9 +43,6 @@ public class ColorBalance {
      * towards warmer temperatures, positive values move towards cooler temperatures.
      */
     public static int getValue() {
-        if (sHasNativeSupport) {
-            return LiveDisplayVendorImpl.getInstance().getColorBalance();
-        }
         return 0;
     }
 
@@ -62,9 +54,6 @@ public class ColorBalance {
      * failed; true in any other case.
      */
     public static boolean setValue(int value) {
-        if (sHasNativeSupport) {
-            return LiveDisplayVendorImpl.getInstance().setColorBalance(value);
-        }
         return false;
     }
 
@@ -73,9 +62,6 @@ public class ColorBalance {
      * @return int
      */
     public static int getMinValue() {
-        if (sHasNativeSupport) {
-            return LiveDisplayVendorImpl.getInstance().getColorBalanceRange().getLower();
-        }
         return 0;
     }
 
@@ -84,9 +70,6 @@ public class ColorBalance {
      * @return int
      */
     public static int getMaxValue() {
-        if (sHasNativeSupport) {
-            return LiveDisplayVendorImpl.getInstance().getColorBalanceRange().getUpper();
-        }
         return 0;
     }
 }
-- 
2.17.0

